Bottom: 07725cd71cb09b2ec69f005346969c06a50da291
Top:    3d12b4c775d154f3958c5ecbb0a8fdc3f4362330
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-08-30 21:51:35 +0800

Refresh of vhost_net-minor-tweak2

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 9f1d796..0e6d43e 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -2494,8 +2494,8 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 		/* FIXME: use u16 for vq->num */
 		copied = min((u16)(vq->num - last_avail_idx), total);
 		ret2 = vhost_copy_from_user(vq, &indices[ret - total],
-			 &vq->avail->ring[last_avail_idx & (vq->num - 1)],
-					copied * sizeof *indices);
+					    &vq->avail->ring[last_avail_idx],
+					    copied * sizeof *indices);
 		if (unlikely(ret2)) {
 			vq_err(vq, "Failed to get descriptors\n");
 			return -EFAULT;
@@ -2506,7 +2506,7 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 	}
 
 	if (!heads)
-		return ret;
+		goto out;
 
 	for (i = 0; i < ret; i++)
 		heads[i].id = indices[i];
@@ -2516,8 +2516,9 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 	while (total) {
 		copied = min((u16)(vq->num - last_used_idx), total);
 		ret2 = vhost_copy_to_user(vq,
-				&vq->used->ring[last_used_idx & (vq->num - 1)],
-				&heads[ret - total], copied * sizeof *used);
+					  &vq->used->ring[last_used_idx],
+					  &heads[ret - total],
+					  copied * sizeof *used);
 
 		if (unlikely(ret2)) {
 			vq_err(vq, "Failed to update used ring!\n");
@@ -2528,9 +2529,9 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 		total -= copied;
 	}
 
+out:
 	/* Only get avail ring entries after they have been exposed by guest. */
 	smp_rmb();
-
 	return ret;
 }
 EXPORT_SYMBOL(vhost_prefetch_desc_indices);
